e2e serving tests fail for operator on s390x

**Describe the bug**
When running the operator e2e tests via prow against a s390x-based cluster all TestKnativeServingDeployment testcases fail. See here for details: https://prow.knative.dev/view/gs/knative-prow/logs/s390x-e2e-tests_operator_main_periodic/1561508625761964032

The tests have been running fine up until 08/17: https://prow.knative.dev/?job=*s390x*operator_main*.
Presumably a change was introduced around that time that breaks the tests on s390x.

When trying to debug into the issue I noticed that upon starting the serving tests the operator pod crashes multiple times with this error:

```
...
{"severity":"INFO","timestamp":"2022-08-22T11:10:16.195556374Z","logger":"knative-operator.manifestival","caller":"manifestival@v0.7.1/manifestival.go:145","message":"Merging","commit":"44f993d-dirty","knative.dev/pod":"knative-operator-765f47dfc8-g24st","diff":"{\"$setElementOrder/webhooks\":[{\"name\":\"validation.webhook.serving.knative.dev\"}],\"metadata\":{\"annotations\":null},\"webhooks\":[{\"name\":\"validation.webhook.serving.knative.dev\",\"rules\":[{\"apiGroups\":[\"autoscaling.internal.knative.dev\",\"networking.internal.knative.dev\",\"serving.knative.dev\"],\"apiVersions\":[\"*\"],\"operations\":[\"CREATE\",\"UPDATE\",\"DELETE\"],\"resources\":[\"metrics\",\"podautoscalers\",\"certificates\",\"ingresses\",\"serverlessservices\",\"configurations\",\"revisions\",\"routes\",\"services\"],\"scope\":\"*\"}]}]}"}
{"severity":"INFO","timestamp":"2022-08-22T11:10:16.196075377Z","logger":"knative-operator.manifestival","caller":"manifestival@v0.7.1/manifestival.go:207","message":"Updating","commit":"44f993d-dirty","knative.dev/pod":"knative-operator-765f47dfc8-g24st","name":"/validation.webhook.serving.knative.dev","type":"admissionregistration.k8s.io/v1, Kind=ValidatingWebhookConfiguration"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf99922]

goroutine 485 [running]:
k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetKind(...)
	k8s.io/apimachinery@v0.23.9/pkg/apis/meta/v1/unstructured/unstructured.go:226
github.com/manifestival/manifestival.okToDelete(0x0)
	github.com/manifestival/manifestival@v0.7.1/manifestival.go:234 +0x42
github.com/manifestival/manifestival.Manifest.delete({{0xc00019aa00, 0x18, 0x20}, {0x1cccd78, 0xc000550210}, {{0x1ccef70, 0xc000999c50}, 0x0}}, 0xc0002fcf38, {0x0, ...})
	github.com/manifestival/manifestival@v0.7.1/manifestival.go:179 +0x5c
github.com/manifestival/manifestival.Manifest.Delete({{0xc00019aa00, 0x18, 0x20}, {0x1cccd78, 0xc000550210}, {{0x1ccef70, 0xc000999c50}, 0x0}}, {0x0, 0x0, ...})
	github.com/manifestival/manifestival@v0.7.1/manifestival.go:116 +0x1e2
knative.dev/operator/pkg/reconciler/common.DeleteObsoleteResources.func1({0x1cccc28, 0xc000b03320}, 0xc000f2e200, {0x1cdee20, 0xc000436000})
	knative.dev/operator/pkg/reconciler/common/stages.go:116 +0x192
knative.dev/operator/pkg/reconciler/common.Stages.Execute(...)
	knative.dev/operator/pkg/reconciler/common/stages.go:37
knative.dev/operator/pkg/reconciler/knativeserving.(*Reconciler).ReconcileKind(0xc000366150, {0x1cccc28, 0xc000b03320}, 0xc000436000)
	knative.dev/operator/pkg/reconciler/knativeserving/knativeserving.go:123 +0x5ba
knative.dev/operator/pkg/client/injection/reconciler/operator/v1beta1/knativeserving.(*reconcilerImpl).Reconcile(0xc00063e8c0, {0x1cccc28, 0xc000b032f0}, {0xc0005e4260, 0x20})
	knative.dev/operator/pkg/client/injection/reconciler/operator/v1beta1/knativeserving/reconciler.go:235 +0x654
knative.dev/pkg/controller.(*Impl).processNextWorkItem(0xc0003c66c0)
	knative.dev/pkg@v0.0.0-20220818004048-4a03844c0b15/controller/controller.go:542 +0x4e8
knative.dev/pkg/controller.(*Impl).RunContext.func3()
	knative.dev/pkg@v0.0.0-20220818004048-4a03844c0b15/controller/controller.go:491 +0x70
created by knative.dev/pkg/controller.(*Impl).RunContext
	knative.dev/pkg@v0.0.0-20220818004048-4a03844c0b15/controller/controller.go:489 +0x3a6
```

The operator-webhook pod shows this error (not sure if it is related, though):

```
{"severity":"ERROR","timestamp":"2022-08-22T11:56:40.548349627Z","logger":"operator-webhook.ConversionWebhook","caller":"controller/controller.go:566","message":"Reconcile error","commit":"44f993d-dirty","knative.dev/pod":"operator-webhook-6dd568bf97-8jmf8","knative.dev/traceid":"75834858-c0ff-4814-bd55-c75ed3b05205","knative.dev/key":"knativeservings.operator.knative.dev","duration":"120.315312ms","error":"failed to update webhook: Operation cannot be fulfilled on customresourcedefinitions.apiextensions.k8s.io \"knativeservings.operator.knative.dev\": the object has been modified; please apply your changes to the latest version and try again","stacktrace":"knative.dev/pkg/controller.(*Impl).handleErr\n\tknative.dev/pkg@v0.0.0-20220818004048-4a03844c0b15/controller/controller.go:566\nknative.dev/pkg/controller.(*Impl).processNextWorkItem\n\tknative.dev/pkg@v0.0.0-20220818004048-4a03844c0b15/controller/controller.go:543\nknative.dev/pkg/controller.(*Impl).RunContext.func3\n\tknative.dev/pkg@v0.0.0-20220818004048-4a03844c0b15/controller/controller.go:491"}
```


**Expected behavior**
The e2e test are running fine, just as they did prior to 08/17.

**To Reproduce**
Run the e2e prow job ''using phaino, e.g.:

```
cd test-infra
mkpj --config-path=config.yaml --job-config-path=prow/jobs/generated/knative/operator-main.gen.yaml --job=continuous_operator_main_periodic > phaino-config/operator
phaino --privileged --timeout 0 phaino-config/operator
```

**Knative release version**
main

--- a/pkg/reconciler/knativeserving/knativeserving.go
+++ b/pkg/reconciler/knativeserving/knativeserving.go
@@ -148,7 +148,7 @@ func (r *Reconciler) transform(ctx context.Context, manifest *mf.Manifest, comp 
 func (r *Reconciler) installed(ctx context.Context, instance base.KComponent) (*mf.Manifest, error) {
 	// Create new, empty manifest with valid client and logger
 	installed := r.manifest.Append()
-	stages := common.Stages{common.AppendInstalled, ingress.AppendInstalledIngresses, r.transform}
+	stages := common.Stages{common.AppendInstalled, ingress.AppendInstalledIngresses, r.filterDisabledIngresses, r.transform}
 	err := stages.Execute(ctx, &installed, instance)
 	return &installed, err
 }

